### PR TITLE
CBG-189: Import not attempted on purged document

### PR DESCRIPTION
**Pull request by JRascagneres**
- [ ] Not to be merged. Change branch to cobalt.

### CODE REVIEW

File: base/error.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ var (
 	ErrIndexAlreadyExists    = &sgError{"Index already exists"}
 	ErrNotFound              = &sgError{"Not Found"}
 	ErrUpdateCancel          = &sgError{"Cancel update"}
+	ErrImportIgnored         = &sgError{"Ignore Import"}
```
Comment (adamcfraser) [inline base/error.go]: I feel like we could make this more precise until we actually have another use case for it - calling it ErrImportCancelledPurged would make it crystal clear why it's being ignored.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/crud.go [hunk 2]
```diff
@@ -1214,6 +1214,8 @@ func (db *Database) updateAndReturnDoc(
 		if err != nil {
 			if err == base.ErrDocumentMigrated {
 				base.DebugfCtx(db.Ctx, base.KeyCRUD, "Migrated document %q to use xattr.", base.UD(key))
+			} else if err == base.ErrImportIgnored {
+				//Do nothing
```
[the remark this answers is not in the document]
Comment (JRascagneres) [reply db/crud.go]: So I actually realized that this is never reached as we catch the error before this function is ran therefore this case has been removed. Do you still want a debug messaged logged but somewhere else instead?

File: rest/api_test.go [hunk 3]
```diff
@@ -3671,6 +3671,50 @@ func TestUnsupportedConfig(t *testing.T) {
...
+	response := rt.SendRequest("GET", "/db/key", "")
+	fmt.Println(response.Body)
+
+	numErrorsAfter, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyErrorCount).String())
+	assert.NoError(t, err)
+
+	assert.Equal(t, numErrors, numErrorsAfter)
+}
+
+func TestRestSettingPurged(t *testing.T) {
```
Comment (adamcfraser) [inline rest/api_test.go:3701]: I'm not sure what scenario this test is covering.  I think the PUT on line 3708 is going to return an error (fields with leading underscore are rejected), so the doc wouldn't be written in the first place.  If you think it's needed for a specific scenario, can you share the details?

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Fix looks good, but a question about one of the new tests.

File: rest/api_test.go (hunk 3 above)
Comment (JRascagneres) [reply rest/api_test.go:3701]: > I'm not sure what scenario this test is covering. I think the PUT on line 3708 is going to return an error (fields with leading underscore are rejected), so the doc wouldn't be written in the first place. If you think it's needed for a specific scenario, can you share the details?



Yes exactly, this is what is was meant to test. . Added this to ensure the change didn't break this original behaviour. I have however altered the test a little now to be more appropriate.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**